cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/DistributedProgramContainerModule.java

```diff
@@ -133,7 +134,7 @@ public DistributedProgramContainerModule(CConfiguration cConf, Configuration hCo
   @Override
   protected void configure() {
     List<Module> modules = getCoreModules();
-
+    modules.add(new NoOpAuditLogModule());
```

Comment:
Any reason not to put inside `getCoreModules`?

Also nit: We tend to create an implementation of io.cdap.cdap.common.runtime.RuntimeModule and have logic for different modes over there. It helps when logic need to change. Not sure if it's needed for this use case, but something to consider